avm/src/daemon/ctrl.rs

```diff
@@ -76,9 +76,32 @@ impl LogRendezvousHash {
     &mut self.sorted_hashes
   }
 
+  pub fn _get_primary_node_id(&self, key: &str) -> &str {
+    self._get_assigned_nodes_id(key)[0]
+  }
+
+  pub fn _get_assigned_nodes_id(&self, key: &str) -> Vec<&str> {
+    let top = 3; // Number of replicas
+    if self.sorted_hashes.len() == 1 {
+      // If only one node, nothing to do
+      return self
+        .sorted_hashes
+        .iter()
+        .map(|hash| &hash.id as &str)
+        .collect::<Vec<&str>>();
+    }
+    let mut idx = self._get_idx_for_key(key);
+    let mut ids: Vec<&str> = Vec::new();
+    for _ in 0..top {
+      ids.push(&self.sorted_hashes[idx].id);
+      idx = (idx + 1) % self.sorted_hashes.len();
+    }
+    ids
+  }
+
   // Runs a binary search for the record whose hash is closest to the key hash without
   // going over. If none are found, the *last* record in the list is returned as it wraps around.
-  pub fn _get_id_for_key(&self, key: &str) -> &str {
+  fn _get_idx_for_key(&self, key: &str) -> usize {
```

Comment:
Functions that are in use (this one as `_get_assigned_nodes_id` above) don't need the prefixed underscore because they are being used, so can we remove that?

Reply:
Yes, of course